core/templates/pages/exploration-editor-page/services/exploration-rights.service.ts

```diff
@@ -148,8 +151,15 @@ angular.module('oppia').factory('ExplorationRightsService', [
             data.rights.voice_artist_names, data.rights.viewer_names,
             data.rights.status, data.rights.cloned_from,
             data.rights.community_owned, data.rights.viewable_if_private);
+          if (onExplorationRightsUpdatedcallback) {
+            onExplorationRightsUpdatedcallback();
+          }
+          explorationUnpublishedEventEmitter.emit();
```

Comment:
Instead of having a callback as well as even an emitter, can we just have one? i.e, just subscribe to the event instead of passing the callback function? [not sure!]

Reply:
No, since the emitter can only be used by one subscribe event. There is already one present in a different file

Reply:
> No, since the emitter can only be used by one subscribe event.

Are you sure about this?

Reply:
callback is used in 
https://github.com/oppia/oppia/blob/96a08a269aff953208e31b90ffcba074629298ad/core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.ts#L340-L349
Subscrive is used in https://github.com/oppia/oppia/blob/96a08a269aff953208e31b90ffcba074629298ad/core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts#L148-L160

Reply:
Is this resolved?